src/main.mo

```diff
@@ -195,6 +204,13 @@ actor class Directory(initialOwner : ?Principal) = self {
       };
     };
 
+    public func assetIdExistsInLedger(assetId : Nat) : async* () {
+      let nLedgerAssets = await ledgerActor.nFtAssets();
+      if (assetId > nLedgerAssets) {
+        throw Error.reject("Asset id does not exist in the ledger");
+      };
+    };
+
```

Comment:
We have to cache `nLedgerAssets` so that we don't have to call the ledger each time. The majority of the additions of new tokens will be for asset ids below the cached value, so we can bypass the call.

Reply:
Could then also expose the cached value in metrics.